docs/advanced/bootloader.md

```diff
@@ -2,14 +2,18 @@
 
 This guide explains how to create a bootloader, how to create a main program to go with the bootloader and how to use this bootloader to perform software updates.
 
+## Porting bootloader
+
+Please follow [the bootloader porting guide](https://docs.mbed.com/docs/mbed-os-handbook/en/latest/advanced/flash/) for adding a bootloader support for your target.
```

Comment:
How about adding: "In order for a target to support Bootloader, the flash on the target needs to support IAP. Porting guide for the FLASH IAP is located here: https://docs.mbed.com/docs/mbed-os-handbook/en/latest/advanced/flash/ . " Then comes all the below i.e. how to create bootloader.. modify the target.restrict_size etc etc.
I feel this gives better context to why the flash IAP needs to be ported rather than starting off directly..